api/routes.py

```diff
@@ -49,6 +50,20 @@ def initialize_circulation_manager():
             # on initial setup) are committed before continuing.
             app.manager._db.commit()
 
+@app.before_first_request
+def initialize_app_settings():
+    _db = app.manager._db
+
+    bearer_token_signing_secrets = [
+        token.value for token in
+        _db.query(ConfigurationSetting).filter(
+            ConfigurationSetting.key==Configuration.BEARER_TOKEN_SIGNING_SECRET
+        ).all()
```

Comment:
I wasn't sure if this should be a list, `first()`, or `one_or_none()`.

Reply:
Couple things, some of which I just figured out:

1. I think there can only be one bearer token signing secret at any given time, because rows in `ConfigurationSetting` have to be [unique for a given combination of key name, external integration id, and library id](https://github.com/NYPL-Simplified/server_core/blob/6c54a9105f530c29389feeba3beb0d9313314355/model/configuration.py#L568-L570), and since this is a sitewide secret, the only operative value is the key name. So you don't have to worry about getting multiple ones by query.
2. The normal way to get the signing secret value is to call [`api.authenticator.BearerTokenSigner.bearer_token_signing_secret()`](https://github.com/NYPL-Simplified/circulation/blob/develop/api/authenticator.py#L2247). However, that method relies on a call to `ConfigurationSetting.sitewide_secret()`, which will [create a randomly generated value](https://github.com/NYPL-Simplified/server_core/blob/6c54a9105f530c29389feeba3beb0d9313314355/model/configuration.py#L625-L628) if the named setting doesn't already exist. That would mean you couldn't actually set the value explicitly if it was unset, since checking for it would also set it.

Given the way it's set up, I think the simplest thing may be to drop the requirement that you be able to set the value from an environment variable, and just make the single call to `BearerTokenSigner.bearer_token_signing_secret(_db)` that you've got below. That'll make sure a value is in there, creating one if necessary.

Reply:
Honestly it's possible this whole thing isn't necessary, since the first time any piece of code tries to retrieve the signing secret it'll get set if it doesn't already exist. That said, I think having an `initialize_app_settings()` function is probably worthwhile, and it's reasonable to be explicit in that function that we want a signing secret to exist.